src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  debounce,
+  addAbsolutePosition,
+  noop,
+} from '../../lib/utils';
+import { Connector, Hits, Hit, FindAnswersResponse } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'answers',
+  connector: true,
+});
+
+export type AnswersRendererOptions = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Hits;
+
+  /**
+   * whether it's still loading the results from the /answers API
+   */
+  isLoading: boolean;
+};
+
+export type AnswersConnectorParams = {
+  /**
+   * Attributes to use for predictions.
+   * If empty, we use all `searchableAttributes` to find answers.
+   * All your `attributesForPrediction` must be part of your `searchableAttributes`.
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: string[];
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { queryLanguages, attributesForPrediction = ['*'], nbHits = 1 } =
+      widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastAnswersResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRenderFn = debounce(renderFn, 200);
```

Comment:
I think the debounce timeout should be a parameter; I can imagine tweaking it around to find the best value

Reply:
Makes sense https://github.com/algolia/instantsearch.js/pull/4581/commits/725347141e15818dc4fa19ddc38d9252593d03ce